src/moonlink_connectors/src/pg_replicate/util.rs

```diff
@@ -304,6 +304,23 @@ fn convert_array_cell(cell: ArrayCell) -> Vec<RowValue> {
                     .unwrap_or(RowValue::Null)
             })
             .collect(),
+        ArrayCell::Composite(values) => values
+            .into_iter()
+            .map(|v| {
+                v.map(|cells| {
+                    let struct_values: Vec<RowValue> = cells
+                        .into_iter()
+                        .map(|cell| PostgresTableRow(TableRow { values: vec![cell] }).into())
+                        .map(|row: MoonlinkRow| {
+                            assert_eq!(row.values.len(), 1, "Composite cell should have exactly one value, but got {} values. This indicates an unexpected conversion from PostgresTableRow to MoonlinkRow", row.values.len());
+                            row.values.into_iter().next().unwrap_or(RowValue::Null)
+                        })
+                        .collect();
+                    RowValue::Struct(struct_values)
+                })
+                .unwrap_or(RowValue::Null)
+            })
+            .collect(),
```

Comment:
Need to add unit test here: https://github.com/Mooncake-Labs/moonlink/blob/586f4ce775be03caa7341b8077d3bb9c30a06adf/src/moonlink_connectors/src/pg_replicate/util.rs#L545